### PR TITLE
Add size preset prop to TextGroup

### PR DESCRIPTION
**Pull request by jshakes**
I spoke with Daniel and confirmed that there are three possible sizes of text groups: small, medium and large. Each size corresponds to a different size of Hed.

This PR adds a `size` prop to the TextGroup component with a default of medium. It also adds a size value to the Hed component.

### CODE REVIEW

File: src/components/BadgeGroup/BadgeGroup.jsx [hunk 1]
```diff
@@ -25,6 +26,7 @@ function BadgeGroup ( {
 			<div className={styles.text}>
 				<TextGroup
 					kicker={kicker}
+					size={size === 'large' ? 'medium' : 'small'}
```
Comment (chriszarate) [inline src/components/BadgeGroup/BadgeGroup.jsx:29]: o_O

Comment (jshakes) [reply src/components/BadgeGroup/BadgeGroup.jsx:29]: I was inspired by Starbucks cup sizing

Comment (jshakes) [reply src/components/BadgeGroup/BadgeGroup.jsx:29]: I'm wondering if t-shirt sizing makes sense for the Badge component; specifying a numeric value seems more intuitive. We could still restrict the possible values if we wanted

File: src/components/BadgeGroup/BadgeGroup.stories.mdx [hunk 2]
```diff
@@ -33,17 +33,57 @@ A `BadgeGroup` is a <LinkTo kind="TextGroup">TextGroup</LinkTo> accompanied by a
...
+### Small with no badge
+
+<Canvas>
+	<Story
+		args={{
+			tagline: 'Quartz is a guide to the new global economy for people in business who are excited by change. We cover business, economics, markets, finance, technology, science, design, and fashion.',
+			title: 'Quartz',
+		}}
+		name="Small with no badge"
+		size="large"
```
Comment (chriszarate) [inline src/components/BadgeGroup/BadgeGroup.stories.mdx:65]:
```suggestion
		size="small"
```

**Pull request merged**